x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.ts

```diff
@@ -58,11 +59,16 @@ export class SingleMetricViewerEmbeddableFactory
       const { resolveEmbeddableSingleMetricViewerUserInput } = await import(
         './single_metric_viewer_setup_flyout'
       );
-      return await resolveEmbeddableSingleMetricViewerUserInput(
+      const userInput = await resolveEmbeddableSingleMetricViewerUserInput(
         coreStart,
         pluginStart,
-        singleMetricServices
+        singleMetricServices.mlApiServices
       );
+
+      return {
+        ...userInput,
+        title: userInput.panelTitle,
```

Comment:
When I edit the visualization, the title always goes back to the default title you get when first adding the embeddable, rather than using the currently saved title.

<img width="1296" alt="Screenshot 2024-04-03 at 11 35 58" src="https://github.com/elastic/kibana/assets/7405507/2b4564e1-0579-4690-9f77-a04d6e22f93e">

Reply:
Fixed in 6b6025afec1fb298ffdbbefda3cc1a830b3c4594